src/OrchardCore/OrchardCore.Recipes.Core/ConfigurationMethodProvider.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Scripting;
+
+
+namespace OrchardCore.Recipes
+{
+    public class ConfigurationMethodProvider : IGlobalMethodProvider
+    {
+        private readonly GlobalMethod _globalMethod;
+
+        public ConfigurationMethodProvider(ShellSettings settings)
+        {
+            _globalMethod = new GlobalMethod
+            {
+                Name = "configuration",
+                Method = serviceprovider => (Func<string, object>)(name =>
+                {
+                    return settings.ShellConfiguration[name];
+                })
```

Comment:
Just for infos `settings[name]` would do the same because it uses internally `_configuration[name]`. And we register `shellSettings.ShellConfiguration` (that is an `IShellConfiguration`) as a tenant level singleton.

**So in this context better to inject `IShellConfiguration configuration` and then just return `configuration[name]`.**

Reply:
Also you could use inline expression instead of create a func block